docs/docs/troubleshooting-common-errors.md

```diff
@@ -0,0 +1,219 @@
+- [gatsby-image](/packages/gatsby-image/): `gatsby-transformer-sharp`, and `gatsby-plugin-sharp`
+
+Rather than packaging up the other dependent libraries alongside these plugins, they can stay smaller in size when they are published and are able to rely on alternative implementations. One example is `gatsby-plugin-sass` that can use either the Node.js or Dart implementations of SASS.
+
+To resolve these errors, identify the packages that haven't been installed, the error message might look like this:
+
+```shell
+...
+success run queries - 0.095s - 8/8 84.63/s
+
+ ERROR #98123  WEBPACK
+
+Generating SSR bundle failed
+
+Can't resolve '@emotion/core' in '/Users/you/tmp/gatsby-site/.cache' // highlight-line
+
+File: .cache/develop-static-entry.js
+```
+
+This error is a result of Gatsby having failing to find `@emotion/core` because `gatsby-plugin-emotion` has been installed and added to the `gatsby-config`, without installing the emotion library. Install it like this:
+
+```shell
+npm install --save @emotion/core
+```
+
+Or replace `@emotion/core` with the name of the library that is missing. With the plugin installed, any necessary libraries installed, and the plugin added to your `gatsby-config`, it should resolve this error.
+
+## Errors in styling
+
+The following errors are related to styles in your site, using CSS, preprocessors, or CSS-in-JSS solutions.
+
+### Inconsistent CSS styles between develop and build using styled-components or emotion
+
+A common problem that trips up users that install and begin to use styled-components or emotion is not including the related plugin in the config. Because `gatsby develop` doesn't run server-side rendering, the build may look different if the plugin is not included to tell Gatsby to server-side render the styles for the CSS-in-JS solution being used.
+
+Adding `gatsby-plugin-styled-components` (in the case of styled-components) or `gatsby-plugin-emotion` (in the case of emotion) to `gatsby-config.js` will inform Gatsby to process the styles server-side so they display correctly in the final build.
+
+## Errors with GraphQL
+
+Gatsby's GraphQL data layer provides access to build time data, there are sometimes errors you may encounter while implementing plugins that are sourcing data or adding nodes to the schema yourself.
+
+### Unknown field 'A' on type 'B'
+
+If the data you are requesting in a GraphQL query differs from what has been [sourced](/docs/content-and-data/) in the GraphQL schema you might encounter an error like `Unknown field 'A' on type 'B'`. Like the error suggests, a field you are asking for is not defined under the type that is listed. If your site is still building okay, you can open up `localhost:8000/___graphql` and examine your schema which includes the definition of what fields are included on the type provided by the error. This can help you identify what fields aren't being created and locate where those fields should be created, whether by a plugin or in your code.
```

Comment:
```suggestion
If the data you are requesting in a GraphQL query differs from what has been [sourced](/docs/content-and-data/) in the GraphQL schema you might encounter an error like `Unknown field 'A' on type 'B'`. As the error suggests, a field you are asking for is not defined under the type that is listed. If your site is still building okay, you can open up `localhost:8000/___graphql` and examine your schema, which includes the definition of what fields are included on the type provided by the error. This can help you identify what fields aren't being created and locate where those fields should be created, whether by a plugin or in your code.
```